lib/gql_serializer.rb

```diff
@@ -46,55 +46,71 @@ def self.query_include(model, hasharray)
     include_array
   end
 
-  def self.serialize(record, hasharray, options)
+  def self.serialize(record, hasharray, options, instructions = {})
 
     if record.nil?
       return nil
     end
 
+
     if record.respond_to? :map
       return record.map do |r|
-        self.serialize(r, hasharray, options)
+        self.serialize(r, hasharray, options, instructions)
       end
     end
 
-    hash = {}
-    model = record.class
-    all_relations = model.reflections.keys
+    id = "#{record.class}, #{hasharray}"
+    instruction = instructions[id]
+    if (!instruction)
+      instruction = {klass: record.class, hasharray: hasharray, relations: [], attributes: []}
+      instructions[id] = instruction
 
-    relations = hasharray.filter do |e|
-      next true if !e.is_a?(String)
```

Comment:
I know this was there before but if you could update this code to avoid using single letter variable names that would make me much more comfortable.

Reply:
Sure.

Reply:
Done